ci/kind/kind-setup.sh

```diff
@@ -517,12 +517,16 @@ function destroy {
       if [[ -e ~/.antrea/.clusters ]]; then
           clean_kind
       fi
-  else
+  elif [[ $CLUSTER_NAME != "*" ]]; then
       kind delete cluster --name $CLUSTER_NAME
+      destroy_external_servers 
+      delete_vlan_subnets
+      delete_networks
+      # Remove the cluster from the list of clusters
+      if [[ -e ~/.antrea/.clusters ]]; then
+          sed -i "/^$CLUSTER_NAME /d" ~/.antrea/.clusters 2>/dev/null || true
+      fi
```

Comment:
this seems fixing another cleanup issue and might introduce new concurrent issue as it doesn't acquire the lock when updating the file.